keel-sql/src/main/kotlin/com/netflix/spinnaker/migrations/EventUidAssigner.kt

```diff
@@ -0,0 +1,73 @@
+package com.netflix.spinnaker.migrations
+
+import com.netflix.spinnaker.keel.persistence.metamodel.Tables.EVENT
+import com.netflix.spinnaker.keel.sql.inTransaction
+import de.huxhorn.sulky.ulid.ULID
+import java.time.LocalDateTime
+import java.time.ZoneOffset.UTC
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import org.jooq.DSLContext
+import org.jooq.Record1
+import org.slf4j.LoggerFactory
+import org.springframework.boot.context.event.ApplicationReadyEvent
+import org.springframework.context.event.EventListener
+
+class EventUidAssigner(private val jooq: DSLContext) : CoroutineScope {
+
+  private val idGenerator = ULID()
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  @EventListener(ApplicationReadyEvent::class)
+  fun onApplicationReady() {
+    launch {
+      while (countEventsWithNoUID() > 0) {
+        var count = 0
+        jooq.inTransaction {
+          select(EVENT.TIMESTAMP)
+            .from(EVENT)
+            .where(EVENT.UID.isNull)
+            .limit(25)
+            .forShare()
```

Comment:
This should as I understand it, make it safe to have this running on multiple instances. The updates are decoupled from the select anyway so it shouldn't clash in the first place.

Reply:
TIL.

Reply:
Why do you say the updates are decoupled though, if they're both running in the same transaction?

Reply:
See comment on the update below